library/std/src/sync/mutex.rs

```diff
@@ -183,8 +183,33 @@ pub struct Mutex<T: ?Sized> {
 
 // these are the only places where `T: Send` matters; all other
 // functionality works fine on a single thread.
+
+/// SAFETY
+///
+/// The `Send` and `Sync` implementations for `Mutex` ensure that it is safe to
+/// share instances of `Mutex` between threads when the protected data is also
+/// thread-safe. The following explains the safety guarantees:
+///
+/// - `Send` is implemented for `Mutex<T>` if and only if `T` is also `Send`.
+///   This guarantees that `Mutex<T>` can be safely transferred between threads,
+///   and `T` can be sent across thread boundaries. This is crucial for allowing
+///   safe access to the protected data from multiple threads.
+///
+/// - `Sync` is implemented for `Mutex<T>` if and only if `T` is `Send`,
+///   since passing around a &Mutex<T> is basically the same as passing a &mut T.
+///   This ensures that `Mutex<T>` can be safely shared between threads
+///   without requiring further synchronization, assuming `T` can be sent across
+///   thread boundaries. It guarantees that multiple threads can safely access the
+///   protected data concurrently without data races.
```

Comment:
This describes `Sync`, so it should be probably moved to `impl Sync for Mutex`.

Also, I'm not a big fan of `is basically the same as` terminology in the docs. I would much prefer to see something like:

> `Mutex<T>` provides mutable access to `T` to one thread at the time, yet still `T` has to be `Send` because it's **not OK** for `Non-Send` structures to be accessed this way. 

> One example of this is `Rc` (non-atomic reference counted smart pointer), which is not `Send`. We can have multiple copies of `Rc` pointing to same heap allocation with non-atomic reference count. `Mutex<Rc<_>>` would only protect **one** copy of `Rc` from shared access, making it vulnerable to data-races.

Reply:
okay